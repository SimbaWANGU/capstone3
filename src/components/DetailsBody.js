/* eslint-disable react/prop-types */
import React from 'react';
import '../styles/DetailsBody.scss';

function DetailsBody({
  rating,
  appearances,
  goalsScored,
  minutesPlayed,
  totalShots,
  shotsOnTarget,
}) {
  const averageRating = `Average Rating: ${rating}`;
  const totalAppearances = `Total Appearances: ${appearances}`;
  const goals = `Goals: ${goalsScored}`;
  const minutes = `Minutes: ${minutesPlayed}`;
  const shots = `Shots: ${totalShots}`;
  const shotsOn = `Shots On: ${shotsOnTarget}`;
  const shotsOff = `Shots Off: ${totalShots - shotsOnTarget}`;

  return (
    <div className="detailsBody">
      <ul>
        <li>
          <p>{averageRating}</p>
        </li>
        <li>
          <p>{totalAppearances}</p>
        </li>
        <li>
          <p>{goals}</p>
        </li>
        <li>
          <p>{minutes}</p>
        </li>
        <li>
          <p>{shots}</p>
        </li>
        <li>
          <p>{shotsOn}</p>
        </li>
        <li>
          <p>{shotsOff}</p>
        </li>
      </ul>
    </div>
  );
}

export default DetailsBody;
