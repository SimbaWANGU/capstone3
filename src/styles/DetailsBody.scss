$fontColor : rgba(2, 2, 200, .9);


.detailsBody {
  display: flex;
  margin: 10px auto auto auto;
  width: 80%;
  background: rgba( 255, 255, 255, 0.15 );
  box-shadow: 0 8px 32px 0 rgba( 31, 38, 135, 0.37 );
  backdrop-filter: blur( 4.5px );
  -webkit-backdrop-filter: blur( 4.5px );
  border-radius: 10px;
  border: 1px solid rgba( 255, 255, 255, 0.18 );

  ul {
    margin: auto;
    display: flex;
    flex-direction: column;
    padding: 0;

    li {
      list-style-type: none;
      margin: 10px auto;
      color: $fontColor;
      font-size: 18px;
    }
  }

  p {
    margin: 0;
  }
}