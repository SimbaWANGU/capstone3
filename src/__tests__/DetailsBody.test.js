import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import DetailsBody from '../components/DetailsBody';

describe('Details page body section tests', () => {
  const testData = {
    rating: 6,
    appearances: 10,
    goalsScored: 1,
    minutesPlayed: 1000,
    totalShots: 10,
    shotsOnTarget: 4,
  };

  it('Component renders as expected', () => {
    const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <DetailsBody
            rating={testData.rating}
            appearances={testData.appearances}
            goalsScored={testData.goalsScored}
            minutesPlayed={testData.minutesPlayed}
            totalShots={testData.totalShots}
            shotsOnTarget={testData.shotsOnTarget}
          />
        </BrowserRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Component is visible to the user', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <DetailsBody
            rating={testData.rating}
            appearances={testData.appearances}
            goalsScored={testData.goalsScored}
            minutesPlayed={testData.minutesPlayed}
            totalShots={testData.totalShots}
            shotsOnTarget={testData.shotsOnTarget}
          />
        </BrowserRouter>
      </Provider>,
    );
    expect(getByTestId('detailsBodyDisplay')).toBeInTheDocument();
  });
});
