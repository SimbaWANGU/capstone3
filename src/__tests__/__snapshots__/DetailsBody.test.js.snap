// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page body section tests Component renders as expected 1`] = `
<div
  className="detailsBody"
  data-testid="detailsBodyDisplay"
>
  <ul>
    <li>
      <p>
        Average Rating: 6
      </p>
    </li>
    <li>
      <p>
        Total Appearances: 10
      </p>
    </li>
    <li>
      <p>
        Goals: 1
      </p>
    </li>
    <li>
      <p>
        Minutes: 1000
      </p>
    </li>
    <li>
      <p>
        Shots: 10
      </p>
    </li>
    <li>
      <p>
        Shots On: 4
      </p>
    </li>
    <li>
      <p>
        Shots Off: 6
      </p>
    </li>
  </ul>
</div>
`;
